### ISSUE TITLE
Can't specify a report file path using the tilde shortcut

### ISSUE THREAD

**Issue by paul-gene** [squizlabs/PHP_CodeSniffer#2802]
When I use the following command

`mkdir ~/foo && vendor/bin/phpcs --report=junit --report-junit="~/foo/phpcs.xml"`

I get the error

`ERROR: The specified junit report file path "~/foo/phpcs.xml" points to a non-existent directory`

But the directory exists when I check using `ls -l ~/`

What am I doing wrong?

**Comment by gsherwood** [comment]
None of the CLI arguments used to set report file paths supported the tilde shortcut properly. They would have worked if you the files you specified already existed, but would always show an invalid directory error if not. I've fixed this now.

I also found and fixed a problem with `--report-junit` and the toher report-specific file path arguments where specifying a directory would throw a PHP error inside of a friendlier PHPCS error.

Thanks for reporting this. The fix will be in 3.5.4
